File: Vindeep-Chaudhari/1636_Sort_Array_by_Increasing_Frequency.cpp
/**
 * Problem Link - https://leetcode.com/problems/sort-array-by-increasing-frequency/submissions/
 */

class Solution {
public:
    struct comp {
        constexpr bool operator()(
            pair<int, int> const& a,
            pair<int, int> const& b)
            const noexcept
        {
            if(a.first == b.first) {
                return a.second < b.second;
            }
            
            return a.first > b.first;
        }
    };
    
    vector<int> frequencySort(vector<int>& nums) {
        unordered_map <int, int> freqMap;
        for(auto num : nums) {
            freqMap[num]++;
        }
        
        priority_queue <pair<int, int> , vector<pair<int, int>>, comp > pq;
        
        for(auto m: freqMap){
            pq.push({m.second, m.first});
        }
        
        int i = 0;
        while(pq.empty() == false) {
            auto popped = pq.top();
            pq.pop();
            
            
            while(popped.first > 0) {
                nums[i] = popped.second;
                i++;
                popped.first--;
            }
        }
        
        return nums;
    }
};

File: Namrata-Sanger/1636. Sort Array by Increasing Frequency.js
/*
Day 20
1636. Sort Array by Increasing Frequency
Link: https://leetcode.com/problems/sort-array-by-increasing-frequency/
Level: Easy
*/

var frequencySort = function(nums) {
    let maps = new Map()
    
    for (let i = 0; i < nums.length; i++) {
        maps.has(nums[i]) ? maps.set(nums[i],maps.get(nums[i]) + 1) : maps.set(nums[i], 1)
    }
    
    let sorted = [...maps.entries()].sort((a,b) => a[1]===b[1] ? b[0] - a[0] : a[1] - b[1])
    
    let ret = []
    sorted.forEach(x=> {
        while(x[1] > 0) {
            ret.push(x[0])
            x[1]--
        }
    })
    return ret
};
